src/_pytest/pytester.py

```diff
@@ -787,6 +787,11 @@ def inline_run(self, *args, plugins=(), no_reraise_ctrlc=False):
 
         :return: a :py:class:`HookRecorder` instance
         """
+        # (maybe a cpython bug?) the importlib cache sometimes isn't updated
+        # properly between file creation and inline_run (especially if imports
+        # are interspersed with file creation)
+        importlib.invalidate_caches()
```

Comment:
JFI: the cache is mtime based (see 8fd5a658e).

Reply:
yeah my guess in why I couldn't repro this was that mtimes are not being written for whatever filesystem driver travis-ci is using when a file was created in a directory 🤷‍♂